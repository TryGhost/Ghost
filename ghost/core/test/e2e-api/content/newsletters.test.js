const {agentProvider, fixtureManager, matchers} = require('../../utils/e2e-framework');

const newsletterSnapshot = {
    id: matchers.anyObjectId,
    uuid: matchers.anyUuid,
    created_at: matchers.anyISODateTime,
    updated_at: matchers.anyISODateTime
};

describe('Newsletters Content API', function () {
    let agent;

    before(async function () {
        agent = await agentProvider.getContentAPIAgent();
        await fixtureManager.init('api_keys', 'newsletters');
        await agent.authenticate();
    });

    it('Can request only active newsletters', async function () {
        await agent.get('/newsletters/')
            .expectStatus(200)
            .matchHeaderSnapshot({
                'content-version': matchers.anyContentVersion,
                etag: matchers.anyEtag
            })
            .matchBodySnapshot({
                newsletters: Array(3).fill(newsletterSnapshot)
            });
    });

    it('Cannot override filters to fetch archived newsletters', async function () {
        await agent.get('/newsletters/?filter=status:archived')
            .expectStatus(200)
            .matchHeaderSnapshot({
                'content-version': matchers.anyContentVersion,
                etag: matchers.anyEtag
            })
            .matchBodySnapshot({
                newsletters: Array(3).fill(newsletterSnapshot)
            });
    });
});
