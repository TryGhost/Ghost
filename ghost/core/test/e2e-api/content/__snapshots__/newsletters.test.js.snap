// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletters Content API Can request only active newsletters 1: [body] 1`] = `
Object {
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 3,
    },
  },
  "newsletters": Array [
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Default Newsletter",
      "sender_email": null,
      "slug": "default-newsletter",
      "sort_order": 0,
      "subscribe_on_signup": true,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "members",
    },
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Daily newsletter",
      "sender_email": "jamie@example.com",
      "slug": "daily-newsletter",
      "sort_order": 1,
      "subscribe_on_signup": false,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "members",
    },
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Weekly newsletter",
      "sender_email": "jamie@example.com",
      "slug": "weekly-newsletter",
      "sort_order": 2,
      "subscribe_on_signup": true,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "members",
    },
  ],
}
`;

exports[`Newsletters Content API Can request only active newsletters 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "public, max-age=0",
  "content-length": "1090",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Accept-Encoding",
  "x-powered-by": "Express",
}
`;

exports[`Newsletters Content API Cannot override filters to fetch archived newsletters 1: [body] 1`] = `
Object {
  "meta": Object {
    "pagination": Object {
      "limit": 15,
      "next": null,
      "page": 1,
      "pages": 1,
      "prev": null,
      "total": 3,
    },
  },
  "newsletters": Array [
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Default Newsletter",
      "sender_email": null,
      "slug": "default-newsletter",
      "sort_order": 0,
      "subscribe_on_signup": true,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "members",
    },
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Daily newsletter",
      "sender_email": "jamie@example.com",
      "slug": "daily-newsletter",
      "sort_order": 1,
      "subscribe_on_signup": false,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "members",
    },
    Object {
      "created_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "description": null,
      "id": StringMatching /\\[a-f0-9\\]\\{24\\}/,
      "name": "Weekly newsletter",
      "sender_email": "jamie@example.com",
      "slug": "weekly-newsletter",
      "sort_order": 2,
      "subscribe_on_signup": true,
      "updated_at": StringMatching /\\\\d\\{4\\}-\\\\d\\{2\\}-\\\\d\\{2\\}T\\\\d\\{2\\}:\\\\d\\{2\\}:\\\\d\\{2\\}\\\\\\.000Z/,
      "uuid": StringMatching /\\[a-f0-9\\]\\{8\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{4\\}-\\[a-f0-9\\]\\{12\\}/,
      "visibility": "members",
    },
  ],
}
`;

exports[`Newsletters Content API Cannot override filters to fetch archived newsletters 2: [headers] 1`] = `
Object {
  "access-control-allow-origin": "*",
  "cache-control": "public, max-age=0",
  "content-length": "1090",
  "content-type": "application/json; charset=utf-8",
  "content-version": StringMatching /v\\\\d\\+\\\\\\.\\\\d\\+/,
  "etag": StringMatching /\\(\\?:W\\\\/\\)\\?"\\(\\?:\\[ !#-\\\\x7E\\\\x80-\\\\xFF\\]\\*\\|\\\\r\\\\n\\[\\\\t \\]\\|\\\\\\\\\\.\\)\\*"/,
  "vary": "Accept-Version, Accept-Encoding",
  "x-powered-by": "Express",
}
`;
